### PR TITLE
Sanitize relation read query

### LINKED ISSUE #18889: RLS between two Authors with "Is Creator" enabled doesn't work for selecting relation.
### Required System information

- Environment: development
- OS: darwin-arm64
- Strapi Version: 4.15.4
- Node/Yarn Version: npm/8.19.4 node/v16.20.1 darwin arm64 workspaces/false
- Edition: Community
- Database: postgres
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When enabled "Is Creator" on READ/WRITE/DELETE permissions of the role. I can see only content I've created in the LIST/VIEW/EDIT interface of the entity in my example "Product Category". Which is correct.
But when I go to another Entity EDIT UI, and try to assign "Product Category" to it as a Relation, I'm able to see the whole list of Product Category entities created by other Authors. Which is incorrect.

### Steps to reproduce the behavior

1. Go to strapi admin panel as Super Admin
2. Create 2 data types: Product, and Category which are connected to each other with One-to-many relation.
3. Go to Settings > Roles > Edit Author role, and set the following permissions to both types: CREATE, READ (Is Creator), WRITE (Is Creator), DELETE (Is Creator).
4. Create 2 "Author" users, User 1, and User 2.
5. Log in as a User 1, and create a "Category 1".
6. Log in as a User 2, and start creating "Product 1", as a relation you will be able to see and select "Category 1" created by User 1.

### Expected behavior

I should be able to select only relations which I've created.

### Screenshots

image
image
image

### PR DESCRIPTION
**Pull request by Bassel17**
### What does it do?

- Make sure to sanitize relation ready query, needed for conditional permissions

### Why is it needed?

- Conditional permissions not working properly with relation field
- see issue #18889 

### How to test it?

- refer to the issue #18889 
- user should only be able to select the relations they created

### Related issue(s)/PR(s)

fix #18889

### CODE REVIEW

**Comment by innerdvations** [comment]
This has broken some tests that probably need to be updated with the new behaviour

File: packages/core/content-manager/server/src/controllers/relations.ts [hunk 1]
```diff
@@ -242,13 +248,32 @@ export default {
...
+        targetField,
+        {
+          fields: ['id'],
+        } as any,
+        {
+          page: ctx.request.query.page,
+          pageSize: ctx.request.query.pageSize,
+        }
+      );
+
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/controllers/relations.ts:270]: Could you wrap the two loadPages queries in a Promise.all ?

File: packages/core/content-manager/server/src/controllers/relations.ts [hunk 2]
```diff
@@ -257,6 +282,7 @@ export default {
         }
       );
 
+      res.results = uniqBy('id', concat(res.results, resWithIds.results));
       ctx.body = res;
     } else {
       const result = await strapi.entityService.load(model, { id }, targetField, queryParams);
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/controllers/relations.ts:288]: shouldn't we also apply the permission query when loading a XToOne relation?

Comment (Bassel17) [reply packages/core/content-manager/server/src/controllers/relations.ts:287]: yes indeed I probably missed this one

File: packages/core/content-manager/server/src/controllers/relations.ts [hunk 3]
```diff
@@ -242,24 +247,58 @@ export default {
...
       );
+      res.results = uniqBy('id', concat(results, resWithOnlyIds.results));
 
       ctx.body = res;
     } else {
-      const result = await strapi.entityService.load(model, { id }, targetField, queryParams);
+      const [resWithOnlyId, res] = await Promise.all([
+        await strapi.entityService.load(model, { id }, targetField, {
+          fields: ['id'],
+        }),
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/controllers/relations.ts:296]: I believe the await is not necessary here, as the promise will be awaited by the promise all

File: packages/core/content-manager/server/src/controllers/relations.ts [hunk 4]
```diff
@@ -242,24 +247,58 @@ export default {
...
+          {
+            ...permissionQuery,
+            ordering: 'desc',
+          } as any,
+          {
+            page: ctx.request.query.page,
+            pageSize: ctx.request.query.pageSize,
+          }
+        ),
+      ]);
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/controllers/relations.ts:283]: I believe the pagination will not properly work here  🤔 

If the permision query filters out entries that are not filtered out in the first loadPages method, pagination might be broken.

An example :
```
-- Page 1
1, 2, 3, 4, 5
1,     3, 4,   , 6, 7

-- Page 2
6, 7, 8, 9, 10
8,  9, 10, 11, 12 <--- id 7 is not available in this page, but it should
```

Comment (Bassel17) [reply packages/core/content-manager/server/src/controllers/relations.ts:283]: indeed, I'll have to rethink that

**Review by Marc-Roig — APPROVED** [review]
Nice job 🚀🚀🚀 
Code looks good to me (did not do an extensive QA)!

**Review by christiancp100 — APPROVED** [review]

**Pull request merged**